### PR TITLE
A53 ARM processor machine modeling

### PR DESCRIPTION
**Pull request by paul-mchugh**
This is the code for converting the AArch64 A53 CPU's machine model from the LLVM machine model to our format.

### CODE REVIEW

File: include/opt-sched/Scheduler/machine_model.h [hunk 1]
```diff
@@ -163,7 +163,7 @@ class MachineModel {
   // Add a new instruction type.
   void AddInstType(InstTypeInfo &instTypeInfo);
   // Add a new issue type.
-  void AddIssueType(IssueTypeInfo &issueTypeInfo);
+  void addIssueType(IssueTypeInfo &issueTypeInfo);
```
Comment (vangthao95) [inline include/opt-sched/Scheduler/machine_model.h:166]: Nitpick: This should be `void addIssueType(IssueTypeInfo &IssueTypeInfo);`

File: lib/Scheduler/machine_model.cpp [hunk 2]
```diff
@@ -205,7 +205,7 @@ void MachineModel::AddInstType(InstTypeInfo &instTypeInfo) {
   instTypes_.push_back(std::move(instTypeInfo));
 }
 
-void MachineModel::AddIssueType(IssueTypeInfo &issueTypeInfo) {
+void MachineModel::addIssueType(IssueTypeInfo &issueTypeInfo) {
```
Comment (vangthao95) [inline lib/Scheduler/machine_model.cpp:208]: Same as above.

File: lib/Wrapper/OptSchedMachineWrapper.cpp [hunk 3]
```diff
@@ -67,10 +76,12 @@ void OptSchedMachineModel::convertMachineModel(
...
     if (mdlName_ == "ARM-Cortex-A7")
-      MMGen = createCortexA7MMGenerator(&dag, this);
+      MMGen = createCortexA7MMGenerator(&Dag, this);
+    else if (ExactCPU == "cortex-a53")
+      MMGen = createCortexA53MMGenerator(&Dag, this);
     else
-      Logger::Error("Could not find machine model generator for target \"%s\"",
-                    mdlName_.c_str());
+      llvm::report_fatal_error(
+          "Could not find machine model generator for target " + mdlName_);
```
Comment (vangthao95) [inline lib/Wrapper/OptSchedMachineWrapper.cpp:84]: I think you can pass `false` to turn off the diagnostic message that states you can report the error to LLVM.
```
      llvm::report_fatal_error(
          "Could not find machine model generator for target " + mdlName_, false);
```

File: include/opt-sched/Scheduler/machine_model.h (hunk 1 above)
Comment (paul-mchugh) [reply include/opt-sched/Scheduler/machine_model.h]: I implemented all the changes you suggested.

**Review by vangthao95 — APPROVED** [review]
LGTM.

**Pull request merged**